bindings/python/edlib.pyx

```diff
@@ -152,3 +154,91 @@ def align(query, target, mode="NW", task="distance", k=-1, additionalEqualities=
     cedlib.edlibFreeAlignResult(cresult)
 
     return result
+
+
+
+## function to output NICE alignments
+## only 'task=="path"' works
+
+def cigarToNiceAlignment(alignResult, query, target, gapSymbol="-"):
```

Comment:
Ok, so the main "news" are that we need position in target to be able to figure out how to correctly construct NICE alignment. This means CIGAR is not enough any more, we also need this position, and it is not so easy to even explain what this position is or where to get it.

So, what you did is, you took the whole alignResult as an argument. I like that, it "hides" what exactly is needed and simplifies things for the user.
However, I would then also change the name of the function, since Cigar is not in the center anymore, and it does not even matter any more that cigar is used as alignment format. So we could rename it to getNiceAlignmentFromAlignResult or alignResultToNiceAlignment or smth like that.

Btw, I suggest you take a look at https://github.com/Martinsos/edlib/blob/master/apps/aligner/aligner.cpp#L331 .
This method does very similar thing, it takes query, target, alignment and position (in CPP but it is similar) and produces NICE alignment. Ok, this method works with alignment while you work with CIGAR, but it is more or less the same thing. Looking at it, it also becomes clear why position is needed. So I would make sure this implementation is aligned (pun intended) with that one.
Btw, one thing I can see from it is that it takes alignment mode (HW, SHW, NW) that was used for alignment. I wonder, do you also need it? Probably you do hm. I think you will want to look into that.

So to summarize: this is great, just rename it to not mention cigar, and probably you will also need to take the alignment mode as parameter -> check that CPP function to see how and why (if needed).

Also, do a check that makes sure CIGAR is set in alignResult, if not throw. And say in comments that alignResult needs to have cigar filled (meaning that task 'path' was used).

Reply:
Thanks for the feedback @Martinsos 

Would you recommend re-implementing the *cpp code `aligner.cpp`? 

> So to summarize: this is great, just rename it to not mention cigar, and probably you will also need to take the alignment mode as parameter -> check that CPP function to see how and why (if needed).

I've renamed the function. I'm not sure about alignment mode as the parameter....we should put this function in the README

```
import edlib
result = edlib.align(query="TAAGGATGGTCCCATTC", target="AAGGGGTCTCATATC", task="path")
edlib.cigarToNiceAlignment(result, query="TAAGGATGGTCCCATTC", target="AAGGGGTCTCATATC")

{'query_aln': 'TAAGGATGGTCCCAT-TC', 'match_aln': '-||||--||||.|||-||', 'target_aln': '-AAGG--GGTCTCATATC'}
```

> Also, do a check that makes sure CIGAR is set in alignResult, if not throw. And say in comments that alignResult needs to have cigar filled (meaning that task 'path' was used).

This I've tried to do as well. 

Let me know :) Thanks for the help!